challenges/FrontEnd/pages/contact_us.html

```diff
@@ -0,0 +1,159 @@
+<html>
+  <head>
+    <meta charset="utf-8" />
+    <title>Your Company</title>
+    <link rel="icon" href="../resources/img/tab_icon.ico" />
+    <link rel="stylesheet" href="../styles/contact_us_style.css" />
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script>
+    <script>
+      //script to retrieve reCAPTCHA site key and set it up
+      var bus = $({});
+      $(document).ready(function() {
+        $.post("validate_captcha.php", { key: "RECAPTCHA_SITE_KEY" }, function(
+          result
+        ) {
+          result = JSON.parse(result);
+          //create script element for reCAPTCHA integration
+          var script = document.createElement("script");
+          script.src =
+            "https://www.google.com/recaptcha/api.js?render=" + result.key;
+          document.getElementsByTagName("head")[0].appendChild(script);
+          bus.trigger("site-key-set", { key: result.key }); //trigger site-key-set event
+        });
+      });
+    </script>
+    <script>
+      var site_key = "";
+      //once the site-key has been set
+      bus.on("site-key-set", function(event, data) {
+        site_key = data.key; //update var so we have access later when form is submitted
+      });
+
+      $(document).ready(function() {
+        //when document finishes loading
+        $("#contact_us_form").submit(function(event) {
+          event.preventDefault(); //stop default submit
+          //check reCAPTCHA
+          grecaptcha.ready(function() {
+            grecaptcha
+              .execute(site_key, {
+                action: "contact_us"
+              })
+              .then(function(token) {
+                var recaptcha_response = document.getElementById(
+                  "recaptcha_response"
+                );
+                recaptcha_response.value = token; //set reCAPTCHA token
+                //send POST request to verify reCAPTCHA
+                $.post(
+                  "validate_captcha.php",
+                  { response: $("#recaptcha_response").val() },
+                  function(result) {
+                    //parse result into JSON object
+                    result = JSON.parse(result);
+                    if (result.success) {
+                      //reCAPTCHA successfull
+                      var f_name = $("#f_name").val();
+                      var l_name = $("#l_name").val();
+                      var email = $("#email").val();
+                      var message = $("#message").val();
+                      //send POST request to submit form
+                      $.post(
```

Comment:
What's stopping me from bypassing all the recaptcha validation and sending data to submit_form.php? Is this secure?

Reply:
When the user submits the form the first thing that happens is recaptcha validation and the request is made to submit_form.php only if the validation was successful. If you are referring to submitting a post request through other means directly to submit_form.php, then, at the moment nothing is stopping you. To mitigate this I can move the validation part to submit_form.php and send the recaptcha token together with the form information. Only processing the information if the validation is successful.

Reply:
I have pushed a version that mitigates this problem.